### DISCUSSION TITLE
Compiling new firmware

### DISCUSSION THREAD

**Discussion by capitaotmn** [moonglow/FlashForge_Marlin#147] [Q&A] [answered]
Hi Everyone.
I am trying to compile a new firmware as I would like to change drivers and add a probe but I am having a few issues.
Managed to compile but when I send it to the printer I get stuck on the flashforge screen. If I resend the pre compiled fw, all works fine again.

What am I doing wrong?

Thanks in advance.

**Comment by GrandWaz00** [comment, marked as answer]
You may have forgotten to encrypt the binary

https://github.com/moonglow/flashforge_firmware_tool

See also:
https://github.com/moonglow/FlashForge_Marlin/discussions/58#discussioncomment-1331778

Windows binary here:
https://github.com/moonglow/FlashForge_Marlin/discussions/139

**Reply by capitaotmn** [reply]
Thanks for your quick answer.
I did not encrypt the binary, seen that post but completly forgot.

Will try again and report back.
